arctic/store/metadata_store.py

```diff
@@ -0,0 +1,229 @@
+
+    def list_symbols(self):
+        return self.distinct('symbol')
+
+    def has_symbol(self, symbol):
+        return self.find_one({'symbol': symbol}) is not None
+
+    def read_metadata(self, symbol, history=False):
+        """
+        Return the metadata saved for a symbol
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        history: `bool`
+            False: Returns current metadata (Default)
+            True: Returns all metadata entries
+
+        Returns
+        -------
+        metadata document if history=False
+        pandas.DateFrame containing timestamps and metadata entries if history=True
+        """
+        if not history:
+            res = self.find_one({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+            return res and res['metadata']
+        find = self.find({'symbol': symbol}, sort=[('start_time', pymongo.ASCENDING)])
+        times = []
+        entries = []
+        for item in find:
+            times.append(item['start_time'])
+            entries.append(item['metadata'])
+        return pd.DataFrame({'metadata': entries}, times)
+
+    def _write_metadata_entry(self, symbol, metadata, start_time):
+        """
+        Create a new metadata entry
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            to be persisted
+        start_time : `datetime.datetime`
+            when entry becomes effective
+
+        Returns
+        -------
+        Document written
+        """
+        document = {'_id': bson.ObjectId()}
+        document['symbol'] = symbol
+        document['metadata'] = metadata
+        document['start_time'] = start_time
+
+        self.insert_one(document)
+
+        logger.debug('Finished writing metadata for %s', symbol)
+
+        return document
+
+    def write_metadata_history(self, collection):
+        """
+        Manually overwrite entire metadata history for symbols in `collection`
+
+        Parameters
+        ----------
+        collection : `dict`
+            {symbol: (list of metadata, list of timestamps)}
+            symbol : `str`
+                symbol name for the item
+            metadata : `dict`
+                to be persisted
+            timestamp : `datetime.datetime`
+                start_time of the corresponding metadata
+            Example:
+                {'example': ([{}], [datetime.utcnow()])
+        """
+        if not isinstance(collection, dict):
+            raise TypeError('collection must be a dictionary.')
+        documents = []
+        for symbol, (entries, times) in collection.items():
+            if len(entries) != len(times):
+                raise ValueError('Number of entries and number of time stamps do not match.')
+            if self.has_symbol(symbol):
+                self.delete_all_metadata(symbol)
```

Comment:
Potentially a dangerous default?  Deleting all metadata is someone is doing a bulk insert of history?